cmake/external/xnnpack.cmake

```diff
@@ -30,15 +32,25 @@ set(FXDIV_SOURCE_DIR ${fxdiv_SOURCE_DIR})
 FetchContent_Declare(pthreadpool URL ${DEP_URL_pthreadpool} URL_HASH SHA1=${DEP_SHA1_pthreadpool})
 onnxruntime_fetchcontent_makeavailable(pthreadpool)
 
+# https://github.com/google/XNNPACK/blob/3b3f7b8a6668f6ab3b6ce33b9f1d1fce971549d1/CMakeLists.txt#L206C82-L206C117
+if(CMAKE_SYSTEM_PROCESSOR MATCHES "^arm64.*" AND NOT CMAKE_C_COMPILER_ID STREQUAL "MSVC")
+  FetchContent_Declare(kleidiai URL ${DEP_URL_kleidiai} URL_HASH SHA1=${DEP_SHA1_kleidiai})
+  onnxruntime_fetchcontent_makeavailable(kleidiai)
+  set(KLEIDIAI_SOURCE_DIR ${kleidiai_SOURCE_DIR})
+endif()
+
+
```

Comment:
Should this be in onnxruntime_external_deps.cmake for consistency/visibility?

Reply:
CMAKE_SYSTEM_PROCESSOR  doesn't work for macOS, since in a Mac build the target processor could be a list( for both x86 and arm64), but CMAKE_SYSTEM_PROCESSOR  cannot be a list.

Reply:
Check Zip-Nuget-Java-Nodejs Packaging Pipeline's build log and see how it worked there.

Reply:
Add the ORT_TARGET_PROCESSOR to determine target processor and fix the exception in Zip_Nuget-Java-Nodejs pipeline